Evo_Game_Function

```diff
@@ -0,0 +1,316 @@
+}
+
+born_Agents <- function(Individual) {
+  parentIndividual <- Individual[Individual["fitness"] == "reproducing", ]
+  if (dim(parentIndividual)[1] > 0) {
+    parentIndividual[, "fitness"] <- "normal"
+    parentIndividual["energy"] <- 0
+    childId <- paste0("A", nrow(Population)+1:nrow(parentIndividual))
+    childIndividual <- data.frame(t = t, ID = childId, parentIndividual[-1:-2])
+    for(i in 1:nrow(childIndividual)) {
+      moveFieldChild <- matrix(NA, 3, 3)
+      moveFieldChild[c(sample(c(1:4, 6:9),1))] <- 1 #exclude 5 to avoid similar position as parent agent.
+      childIndividual[i, c("row.d","col.d")] <- which(moveFieldChild == 1, arr.ind = TRUE)
+      childIndividual[i, c("row.d","col.d")] <- childIndividual[i,c("row.d","col.d")] - c(2,2) #
+    }
+    childIndividual[, c("row","col")] <- childIndividual[, c("row","col")] + childIndividual[, c("row.d","col.d")]
+    childIndividual[, c("row","col")] <- cbind((childIndividual["row"] - 1) %% nrow + 1, (childIndividual["col"] - 1) %% ncol + 1)
+    childIndividual["ind"] <- as.vector(apply(childIndividual[c("row","col")],1, function(x) indField[x[1],x[2]]))
+    Individual[Individual[["ID"]] %in% parentIndividual[["ID"]], ] <- parentIndividual
+    childPopulation <- data.frame(ID = childIndividual["ID"], Birth_Ind = childIndividual["ind"], Birth_t = t, Death_Ind = NA, Death_t = NA, Live_Duration = NA,Parent_ID = parentIndividual["ID"], Food_Consumed = 0)
+    Population <- rbind(Population, childPopulation)
+    Individual <- rbind(Individual, childIndividual)
+    nAgent <- nAgent + nrow(childIndividual)
+    born <- born + nrow(childIndividual)
+  }
+  Population <- Population
+  Individual <- Individual
+  nAgent <- nAgent
+  born <- born
+  return(list(Individual = Individual, Population = Population, nAgent = nAgent, born = born))
+}
+
+moving_Agents <- function(Individual) {
+  normalIndividual <- priorIndividual[priorIndividual["fitness"] == "normal", ]
+  if (dim(normalIndividual)[1] > 0) {
+    normalIndividual["t"] <- t
+    for(i in 1:nrow(normalIndividual)) {
+      moveField <- matrix(NA, 3, 3)
+      moveField[c(sample(c(1:9),1))] <- 1
+      normalIndividual[i, c("row.d","col.d")] <- which(moveField == 1, arr.ind = TRUE)
+      normalIndividual[i, c("row.d","col.d")] <- normalIndividual[i,c("row.d","col.d")] - c(2,2)
+    }
+    normalIndividual[, c("row","col")] <- normalIndividual[, c("row","col")] + normalIndividual[, c("row.d","col.d")]
+    normalIndividual[, c("row","col")] <- cbind((normalIndividual["row"] - 1) %% nrow + 1, (normalIndividual["col"] - 1) %% ncol + 1)
+    normalIndividual["ind"] <- as.vector(apply(normalIndividual[c("row","col")],1, function(x) indField[x[1],x[2]]))
+    Individual[Individual[["ID"]] %in% normalIndividual[["ID"]], ] <- normalIndividual
+  }
+  return(Individual)
+}
+
+distribute_Overlaying_Agents <- function (Individual) {
```

Comment:
In the first part of your code, you included comprehensive comments that guided the reader through your code. I would really appreciate these comments in this part of the code, too. :)